mpitest/test_parallel_record.py

```diff
@@ -0,0 +1,56 @@
+""" Testing out recorders under MPI."""
+
+import os
+import unittest
+
+from openmdao.core import Problem
+from openmdao.core.mpi_wrap import MPI
+from openmdao.recorders import DumpRecorder
+from openmdao.test.simple_comps import FanInGrouped
+from openmdao.test.mpi_util import MPITestCase
+
+if MPI:
+    from openmdao.core.petsc_impl import PetscImpl as impl
+else:
+    from openmdao.core import BasicImpl as impl
+
+
+class TestDumpRecorder(MPITestCase):
+
+    N_PROCS = 2
+
+    def setUp(self):
+        if MPI:
+            if MPI.COMM_WORLD.rank == 0:
+                self.filename = 'data_0.dmp'
+            elif MPI.COMM_WORLD.rank == 1:
+                self.filename = 'data_1.dmp'
+        else:
+            self.filename = 'data.dmp'
+
+    def tearDown(self):
+        if os.path.exists(self.filename):
+            os.remove(self.filename)
+
+    def test_dump_converge_diverge_par(self):
+
+        prob = Problem(impl=impl)
+        prob.root = FanInGrouped()
+
+        rec = DumpRecorder(out='data.dmp')
```

Comment:
Ya.. what I meant was something like this, to avoid the disconnect between the filename in setUP() and the DumpRecorder arg:

    filename = 'data.dmp'
    rec = DumpRecorder(out=filename)

    if MPI:
        filename = 'data_'+MPI.COMM_WORLD.rank+'.dmp'

    with open(filename, 'r') as dumpfile:
        dump = dumpfile.readlines()